x-pack/legacy/plugins/apm/server/routes/index_pattern.ts

```diff
@@ -4,26 +4,14 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Boom from 'boom';
-import { InternalCoreSetup } from 'src/core/server';
 import { getAPMIndexPattern } from '../lib/index_pattern';
+import { createRoute } from './create_route';
 
-const defaultErrorHandler = (err: Error & { status?: number }) => {
-  // eslint-disable-next-line
-  console.error(err.stack);
-  throw Boom.boomify(err, { statusCode: err.status || 500 });
-};
-
-export function initIndexPatternApi(core: InternalCoreSetup) {
-  const { server } = core.http;
-  server.route({
-    method: 'GET',
-    path: '/api/apm/index_pattern',
-    options: {
-      tags: ['access:apm']
-    },
-    handler: async req => {
-      return await getAPMIndexPattern(server).catch(defaultErrorHandler);
-    }
-  });
-}
+export const indexPatternRoute = createRoute(core => ({
+  method: 'GET',
+  path: '/api/apm/index_pattern',
+  handler: async () => {
```

Comment:
This api should also accept the `_debug` flag (there is a setting which will call any apm endpoint with `_debug=true`. I assume this would fail with that enabled.

Reply:
Aye, will address this as well in the final version.